typedoc.js

```diff
@@ -1,23 +1,26 @@
-const { readdirSync } = require("fs");
+const { readdirSync, existsSync } = require("fs");
 
 const constructsDir = "src/constructs";
 const utilsDir = "src/utils";
 
 function getEntryPointsFromSubdirectories(directory) {
   return readdirSync(directory, { withFileTypes: true })
     .filter((dir) => dir.isDirectory())
-    .map(({ name }) => `${directory}/${name}/index.ts`);
+    .map(({ name }) => `${directory}/${name}/index.ts`)
+    .filter((file) => existsSync(file));
 }
 
 module.exports = {
+  exclude: "src/**/*.test.ts",
   entryPoints: [
-    "src/index.ts",
+    "src/patterns/index.ts",
```

Comment:
This solves https://github.com/guardian/cdk/pull/972.

![image](https://user-images.githubusercontent.com/836140/155967682-e1cf1ea6-6c45-42ba-ab39-52959102a63f.png)